### PR TITLE
use 'native-tls' instead of rustls for reqwest

### PR DESCRIPTION
**Pull request by SteiMi**
This aims to solve #36.

Instead of using 'rustls-tls' feature of reqwest, this uses 'native-tls'. This checks the certificates trusted by the operating system, which allows the user to easily trust custom self-signed certificates.

This built fine for me locally on Windows. Let's hope that the CI can build this successfully for all platforms.

Note: I also tried the feature 'rustls-tls-native-roots' which should theoretically also take certificates trusted by the OS into consideration, but I could not get it to trust my certificate.

### CODE REVIEW

**Comment by McPatate** [comment]
Is what you're looking to achieve not possible with `rust-tls`?

**Comment by FileMagic** [comment]
Before reading anything here just know I am a rust noob. I am just curious.

> Is what you're looking to achieve not possible with `rust-tls`?

I think it is possible with `rust-tls` after some cursory glances at some docs.

I was just coming around to this for I saw that this repo is the one related to this issue [here](https://github.com/huggingface/llm-vscode/issues/78#issuecomment-1744688907) in llm-vscode. I think after reading this could instead be usage of [rustls-native-certs](https://github.com/rustls/rustls-native-certs?tab=readme-ov-file) instead of using `native-tls`.

Reading through the reqwests docs [here](https://docs.rs/reqwest/latest/reqwest/tls/#default-tls) tells me that the usage of [native-tls](https://docs.rs/native-tls/latest/native_tls/) will probs give us the best by platform import of tls certs trusted by the os.

However, what is the reasoning behind using `rust-tls` over `native-tls`? I don't see a reason to not use `native-tls` outside of the [openssl compiling dependency](https://github.com/seanmonstar/reqwest?tab=readme-ov-file#requirements)?


TLDR: I think this MR is just to support using tls certs trusted by the host os, and I wonder if the reason they want to use `native-tls` is to just support tls certs by default on this host instead of having to disable tls which is what most people are doing.

**Comment by FileMagic** [comment]
Responding to myself here, I realized that you said in the https://github.com/huggingface/llm-ls/issues/36#issuecomment-1771185733 that you had issues with compiling using openssl.

So I think it would be more reasonable to switch to the usage of [rustls-native-certs](https://github.com/rustls/rustls-native-certs?tab=readme-ov-file) instead which exposes `pub fn load_native_certs() -> Result<Vec<pki_types::CertificateDer<'static>>, std::io::Error>`. I may write up a PR based on this here soon then to get a fix out for this issue.

**Comment by McPatate** [comment]
Closing in favor of #77
